src/EnergyPlus/SQLiteProcedures.hh

```diff
@@ -56,10 +56,10 @@ public:
 	// This allows for testing of private methods in SQLite
 	friend class SQLiteFixture;
 
-	void addScheduleData( int const number, std::string const name, std::string const type, double const minValue, double const maxValue );
+	void addScheduleData( int const number, std::string const & name, std::string const & type, double const minValue, double const maxValue );
```

Comment:
@mbadams5 So I looked at SQLiteProcedures and I think the issue is related to the structures holding references. When you do that you are expecting the lifetime of the referenced object to persist during the lifetime of the referring object. When you pass a literal string to a `std::string const &` its lifetime ends when the function returns, causing the observed problems. By giving turning those arguments into named variables you extend their lifetimes to the end of their scopes: maybe that was long enough for the SQL purposes in the unit tests but this seems like a potential bug exposure "in the wild" and maybe a hard-to-use-safely interface.

I could review this design with you if you share my concern that it might be non-robust. I'd need to understand the SQL approach and what the lifetimes need to be (assuming the write can't happen immediately) and whether avoiding holding copies was done for performance reasons or because later changes to variables really need to be applied when the write happens. In addition to the reference members, the shared pointers and heap allocations outside of a polymorphic/factory context are something that I would normally want to look at. I have the "mandate" to do some code reviews so if @Myoldmopar and @lgentile feel this would be worthwhile then I just need some background from you or @kbenne to get started. If the code is considered settled/immutable at this point then that is fine: we can wait to see if lifetime or performance issues arise down the road.